java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/h2/JdbcToArrowArrayTest.java

```diff
@@ -0,0 +1,372 @@
+      } catch (Exception e) {
+        e.printStackTrace();
+        throw e;
+      }
+
+      assertEquals(rowCount, vector.getRowCount());
+
+      testIntegerList((ListVector) vector.getVector(INT_ARRAY_FIELD_NAME), rowCount, intArrays);
+      testFloatList((ListVector) vector.getVector(FLOAT_ARRAY_FIELD_NAME), rowCount, floatArrays);
+      testStringList((ListVector) vector.getVector(STRING_ARRAY_FIELD_NAME), rowCount, strArrays);
+    }
+  }
+
+  @Test
+  public void testJdbcToArrowWithNulls() throws Exception {
+    int rowCount = 3;
+    Integer[][] intArrays = new Integer[rowCount][];
+    intArrays[0] = null;
+    intArrays[1] = new Integer[] { 0 };
+    intArrays[2] = new Integer[] { 1 };
+
+    Float[][] floatArrays = new Float[rowCount][];
+    floatArrays[0] = new Float[] { 2.0f };
+    floatArrays[1] = null;
+    floatArrays[2] = new Float[] { 3.0f };
+
+    String[][] stringArrays = new String[rowCount][];
+    stringArrays[0] = new String[] { "4" };
+    stringArrays[1] = null;
+    stringArrays[2] = new String[] { "5" };
+
+    insertRows(rowCount, intArrays, floatArrays, stringArrays);
+
+    final JdbcToArrowConfigBuilder builder =
+        new JdbcToArrowConfigBuilder(new RootAllocator(Integer.MAX_VALUE), JdbcToArrowUtils.getUtcCalendar(), false);
+    builder.setArraySubTypeByColumnNameMap(arrayFieldMapping);
+
+    final JdbcToArrowConfig config = builder.build();
+
+    try (ResultSet resultSet = conn.createStatement().executeQuery(QUERY)) {
+      final VectorSchemaRoot vector;
+
+      try {
+        vector = JdbcToArrow.sqlToArrow(resultSet, config);
+      } catch (Exception e) {
+        e.printStackTrace();
+        throw e;
+      }
+
+      assertEquals(rowCount, vector.getRowCount());
+
+      testIntegerList((ListVector) vector.getVector(INT_ARRAY_FIELD_NAME), rowCount, intArrays);
+      testFloatList((ListVector) vector.getVector(FLOAT_ARRAY_FIELD_NAME), rowCount, floatArrays);
+      testStringList((ListVector) vector.getVector(STRING_ARRAY_FIELD_NAME), rowCount, stringArrays);
+    }
+  }
+
+  private void testIntegerList(ListVector listVector, int rowCount, Integer[][] expectedValues) {
+    IntVector vector = (IntVector) listVector.getDataVector();
+    ArrowBuf offsetBuffer = listVector.getOffsetBuffer();
+
+    int prevOffset = 0;
+    for (int row = 0; row < rowCount; ++row) {
+      int offset = offsetBuffer.getInt((row + 1) * ListVector.OFFSET_WIDTH);
+
+      if (expectedValues[row] == null) {
+        assertEquals(0, offset - prevOffset);
+        continue;
+      }
+
+      assertEquals(expectedValues[row].length, offset - prevOffset);
+
+      for (int i = prevOffset; i < offset; ++i) {
+        assertEquals(expectedValues[row][i - prevOffset].intValue(), vector.get(i));
+      }
+
+      prevOffset = offset;
+    }
+  }
+
+  private void testFloatList(ListVector listVector, int rowCount, Float[][] expectedValues) {
+    Float4Vector vector = (Float4Vector) listVector.getDataVector();
```

Comment:
there seems to be a lot of redundant code in these test methods is it possible to refactor to use a lambda to avoid the duplication?

Reply:
The redundancy is due to the different vectors having different types.  I'm not at all familiar with Java lambdas, but from what I'm reading in https://medium.freecodecamp.org/learn-these-4-things-and-working-with-lambda-expressions-b0ab36e0fffc it looks like I can't use it to hide the types?

Reply:
No they can't really hide types, but if the function took two Object values you could have a specific one that does the casts internally for the listVector and expectedValues.  

No need to implement this just a thought.